### ISSUE TITLE
Add API for account deletion

### ISSUE THREAD

**Issue by devgeeks** [SpiderOak/crypton#412]
Users should be able to delete their account, therefore there needs to be an API to surface this to apps using Crypton.

**Comment by daviddahl** [comment]
yes, we need this! it should also cascade to delete all of the user's data as well.

**Comment by devgeeks** [comment]
:+1:

**Comment by grempe** [comment]
:+1: 

This would also allow for before/after actions in the test suite which would buildup/teardown test accounts or other data for each test.  The way it currently is most server tests have dependencies on other previous tests and the test data they create.  This not only prevents tests from being run in isolation, it potentially allows issues to creep in based on the order of test runs both of which are a big no-no.  Any individual test, or suite of tests for a model or set of routes, should be able to be run in isolation with no test data assumptions.

**Comment by grempe** [comment]
I have some work in progress on this issue.
